File: ApplicationCore/Entities/ProductRequestLine.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    //có thể từ nhiều phiếu
    public class ProductRequestLine : BaseEntity
    {
        /// <summary>
        /// meterial product
        /// </summary>
        public Guid? ProductId { get; set; }
        public Product Product { get; set; }

        public Guid? ProductUOMId { get; set; } 
        public UoM ProducUOM { get; set; }

        public Guid RequestId { get; set; }
        public ProductRequest Request { get; set; }

        public Guid? SaleOrderLineId { get; set; }
        public SaleOrderLine SaleOrderLine { get; set; }

        public decimal ProductQty { get; set; }

        public int Sequence { get; set; }
    }
}

File: TMTDentalAPI/ClientApp/src/app/sale-orders/product-request-line.ts
export class ProductRequestLineBasic {
    id: string;
    productId: string;
    product: Object;
    productUOMId: string;
    productUOM: Object;
    saleOrderLineId: string;
    saleOrderLine: Object;
    productQty: number;
    sequence: number;
}

export class ProductRequestLineDisplay {
    id: string;
    productId: string;
    product: Object;
    productUOMId: string;
    productUOM: Object;
    saleOrderLineId: string;
    saleOrderLine: Object;
    productQty: number;
    sequence: number;
}

export class ProductRequestLineSave {
    id: string;
    productId: string;
    productUOMId: string;
    saleOrderLineId: string;
    productQty: number;
    sequence: number;
}